ykcompile/src/lib.rs

```diff
@@ -598,6 +600,21 @@ impl<TT> TraceCompiler<TT> {
         Location::Register(reg)
     }
 
+    fn c_len(&mut self, p: &Place) -> Location {
+        let dst = self.create_temporary();
+        let (loc, _) = self.place_to_location(p, true);
+        match loc {
+            Location::Register(src) | Location::Addr(src) => {
```

Comment:
Can the `Location::Register` case arise for a fat pointer?

Reply:
To be honest, I'm not sure. If `Len` is always preceded by a `Deref`, then no. But if we have a non-referenced array, and use `Len` on it, we might just get a `Register`. In that case, we could possibly even get a `Mem`, but we'll have to implement arrays first in order to test this.

Reply:
So shall we remove `Location::Register` and let it crash? Then we can decide what to do later.

Reply:
Let's do it.

Reply:
Fixed in 70227c9